### PR TITLE
update to diffusers 0.15 and fix code for name changes

### PR DESCRIPTION
**Pull request by lstein**
- This is a port of #3184 to the main branch

### CODE REVIEW

**Comment by GreggHelt2** [comment]
Any chance we could increment this to PR to diffusers 0.15.1?
[v0.15.1: Patch Release to fix safety checker, config access and uneven scheduler](https://github.com/huggingface/diffusers/releases/tag/v0.15.1)

File: invokeai/backend/stable_diffusion/diffusion/cross_attention_control.py [hunk 1]
```diff
@@ -408,7 +408,7 @@ def override_cross_attention(model, context: Context, is_running_diffusers=False
 def get_cross_attention_modules(
     model, which: CrossAttentionType
 ) -> list[tuple[str, InvokeAICrossAttentionMixin]]:
-    from ldm.modules.attention import CrossAttention  # avoid circular import
+    from ldm.modules.attention import CrossAttention  # avoid circular import - TODO: rename as in diffusers?
```
[the remark this answers is not in the document]
Comment (lstein) [reply invokeai/backend/stable_diffusion/diffusion/cross_attention_control.py]: Glad you noticed this. It was leftover code from v2.3. It is now removed. I don't think CrossAttention is needed.

**Comment by lstein** [comment]
> Any chance we could bump this PR to diffusers v0.15.1?
> [v0.15.1: Patch Release to fix safety checker, config access and uneven scheduler](https://github.com/huggingface/diffusers/releases/tag/v0.15.1)

Yes, done.

**Comment by GreggHelt2** [comment]
> > Any chance we could bump this PR to diffusers v0.15.1?
> > [v0.15.1: Patch Release to fix safety checker, config access and uneven scheduler](https://github.com/huggingface/diffusers/releases/tag/v0.15.1)
> 
> Yes, done.

Thanks Lincoln! There are a few new improvements to ControlNet support in v0.15.1 that I really want to utilize for InvokeAI.

**Review by ankit-gautam23 — APPROVED** [review]
LGTM

**Comment by lstein** [comment]
@GreggHelt2 , @JPPhoto and/or @blessedcoolant. This update to diffusers 0.15.1 needs a code review and approval.

**Review by blessedcoolant — APPROVED** [review]

**Pull request merged**